packages/edit-widgets/src/components/edit-widgets-initializer/index.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/compose';
+import { useEffect } from '@wordpress/element';
+import { withDispatch, withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import Layout from '../layout';
+
+function EditWidgetsInitializer( { widgetAreas, setupWidgetAreas } ) {
+	useEffect( () => {
+		if ( ! widgetAreas ) {
+			return;
+		}
+		setupWidgetAreas( widgetAreas );
```

Comment:
Basically to initialize the screen, we need to fetch the widgets and parse them as blocks. Instead of relying on a component being rendered and a result of `widgetAreas` being available here. Why not instead have a unique `setupWidgetAreas` action without any arguments that performs the fetch done here https://github.com/WordPress/gutenberg/pull/15074/files#diff-666bce58b2f8096cdabbd6d570553a4fR25 (using controls) and continues with the rest.

It feels a bit better than having this behavior split between a selector, a component re-rendering and then a dispatch.

Reply:
This control would be helpful here https://github.com/WordPress/gutenberg/pull/15435/files#diff-9fb42816127c9fa3299026bdc8a283dfR159

Reply:
Suggestion applied 👍